src/Collection/CollectionTrait.php

```diff
@@ -540,13 +539,12 @@ public function isEmpty()
         return iterator_count($this->take(1)) === 0;
     }
 
+
     /**
-     * Returns the closest nested iterator that can be safely traversed without
-     * losing any possible transformations.
+     * {@inheritDoc}
      *
-     * @return \Iterator
      */
-    protected function _unwrap()
+    public function _unwrap()
```

Comment:
Shouldnt we keep this method and rather make a new wrapper `public function unwrap()` around it? Would be less akward looking.